tests/templates/test_subroutines/test_select.py

```diff
@@ -0,0 +1,277 @@
+            qml.SELECT(ops, control_wires=control_wires)
+
+        select_ops = recorder.expand().operations
+
+        assert [op.name for op in select_ops] == [op.name for op in expected_gates]
+        assert [op.wires for op in select_ops] == [op.wires for op in expected_gates]
+
+
+class TestErrorMessages:
+    """Test that the correct errors are raised"""
+
+    @pytest.mark.parametrize(
+        ("ops", "control_wires", "msg_match"),
+        [
+            (
+                [qml.PauliX(wires=1), qml.PauliY(wires=0), qml.PauliZ(wires=0)],
+                [1, 2],
+                "Control wires should be different from operation wires.",
+            ),
+            (
+                [qml.PauliX(wires=2)] * 4,
+                [1, 2, 3],
+                "Control wires should be different from operation wires.",
+            ),
+            (
+                [qml.PauliX(wires="a"), qml.PauliY(wires="b")],
+                ["a"],
+                "Control wires should be different from operation wires.",
+            ),
+        ],
+    )
+    def test_control_wires_in_ops(self, ops, control_wires, msg_match):
+        """Test an error is raised when a control wire is in one of the ops"""
+        with pytest.raises(ValueError, match=msg_match):
+            qml.SELECT(ops, control_wires)
+
+    @pytest.mark.parametrize(
+        ("ops", "control_wires", "msg_match"),
+        [
+            (
+                [qml.PauliX(wires=0), qml.PauliY(wires=0), qml.PauliZ(wires=0)],
+                [1],
+                r"Not enough control wires \(1\) for the desired number of operations \(3\). At least 2 control wires required.",
+            ),
+            (
+                [qml.PauliX(wires=0)] * 10,
+                [1, 2, 3],
+                r"Not enough control wires \(3\) for the desired number of operations \(10\). At least 4 control wires required.",
+            ),
+            (
+                [qml.PauliX(wires="a"), qml.PauliY(wires="b"), qml.PauliZ(wires="c")],
+                [1],
+                r"Not enough control wires \(1\) for the desired number of operations \(3\). At least 2 control wires required.",
+            ),
+        ],
+    )
+    def test_too_many_ops(self, ops, control_wires, msg_match):
+        """Test that error is raised if more ops are requested than can fit in control wires"""
+        with pytest.raises(ValueError, match=msg_match):
+            qml.SELECT(ops, control_wires)
+
+
+def select_rx_circuit(angles):
+    """Circuit that uses SELECT for tests."""
+    qml.SELECT([qml.RX(angles[0], wires=[1]), qml.RY(angles[1], wires=[1])], control_wires=0)
+    return qml.expval(qml.PauliZ(wires=1))
+
+
+def manual_rx_circuit(angles):
+    """Circuit that manually creates SELECT for tests."""
+    qml.ctrl(qml.RX(angles[0], wires=[1]), control=0, control_values=0)
+    qml.ctrl(qml.RY(angles[1], wires=[1]), control=0)
+    return qml.expval(qml.PauliZ(wires=1))
+
+
+class TestInterfaces:
```

Comment:
It might be nice to have a test to check that all the interfaces give the same result as well. (not a blocker)